quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -54,6 +55,124 @@ pub enum LeafType {
     Text(QuickwitTextOptions),
 }
 
+fn value_to_pretokenized<T: ToString>(val: T) -> PreTokenizedString {
```

Comment:
is there a discussion somewhere about why we are going for the string route? I think we discussed it but forgot the rationale.

Reply:
I don't think we had that discussion. I based my work on
> For numeric types or dates, we should coerce them to string.

(from the ticket).
The rational I see would be that if we want to do point search (not range &co), casting everything to string is the simplest way, and work rather well. Maybe we could leverange json field instead, but I don't think there is much benefit, and you end up doing twice as many requests to storage when you don't know if a term is an integer or a string.